jt-shadedrelief/src/main/java/it/geosolutions/jaiext/shadedrelief/ShadedReliefDescriptor.java

```diff
@@ -0,0 +1,170 @@
+/* JAI-Ext - OpenSource Java Advanced Image Extensions Library
+ *    http://www.geo-solutions.it/
+ *    Copyright 2018 GeoSolutions
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package it.geosolutions.jaiext.shadedrelief;
+
+import com.sun.media.jai.util.AreaOpPropertyGenerator;
+import java.awt.RenderingHints;
+import java.awt.image.RenderedImage;
+import javax.media.jai.JAI;
+import javax.media.jai.OperationDescriptorImpl;
+import javax.media.jai.ParameterBlockJAI;
+import javax.media.jai.PropertyGenerator;
+import javax.media.jai.ROI;
+import javax.media.jai.RenderedOp;
+import javax.media.jai.registry.RenderedRegistryMode;
+
+/**
+ * An <code>OperationDescriptor</code> describing the "ShadedRelief" operation.
+ *
+ */
+public class ShadedReliefDescriptor extends OperationDescriptorImpl {
+
+    public static final double DEFAULT_AZIMUTH = 315;
+
+    public static final double DEFAULT_ALTITUDE = 45;
+
+    public static final double DEFAULT_Z = 100000;
+
+    private static final double DEGREES_TO_METERS = 111120;
+
+    public static final double DEFAULT_SCALE = DEGREES_TO_METERS;
+
+    /** serialVersionUID */
+    private static final long serialVersionUID = 1L;
+
+    /**
+     * The resource strings that provide the general documentation and specify the parameter list
+     * for a ShadedRelief operation.
+     */
+    private static final String[][] resources = {
+        {"GlobalName", "ShadedRelief"},
+        {"LocalName", "ShadedRelief"},
+        {"Vendor", "it.geosolutions.jaiext"},
+        {"Description", "desc"},
+        {"Version", "ver"},
+        {"arg0Desc", "Region of interest"},
+        {"arg1Desc", "Input NoData"},
+        {"arg2Desc", "Destination NoData"},
+        {"arg3Desc", "X resolution"},
```

Comment:
Need some better description of the following params (here, or at least, somewhere):
* x resolution and y resolution (and I have a hunch of what they do, but best not having to guess)
* vertical exxageration
* the unit scale ratio (believe it's for data in degrees, again, need some explanation)
* altitude, azimuth (of the sun? in what unit? degrees, radians?)